src/pymor/models/iosys.py

```diff
@@ -388,6 +137,28 @@ def __init__(self, A, B, C, D=None, E=None, cont_time=True,
                          error_estimator=error_estimator, visualizer=visualizer, name=name)
         self.__auto_init(locals())
 
+        K = lambda s: s * self.E - self.A
+        B = lambda s: self.B
+        C = lambda s: self.C
+        D = lambda s: self.D
+        dK = lambda s: self.E
+        dB = lambda s: ZeroOperator(self.B.range, self.B.source)
+        dC = lambda s: ZeroOperator(self.C.range, self.C.source)
+        dD = lambda s: ZeroOperator(self.D.range, self.D.source)
+        parameters = Parameters.of(self.A, self.B, self.C, self.D, self.E)
+
+        self.transfer_function = FactorizedTransferFunction(
+            self.dim_input, self.dim_output,
+            K, B, C, D, dK, dB, dC, dD,
+            parameters=parameters, cont_time=cont_time, name=self.name + '_transfer_function')
+
+        self.eval_tf = self.transfer_function.eval_tf
+        self.eval_dtf = self.transfer_function.eval_dtf
+        self.freq_resp = self.transfer_function.freq_resp
+        self.bode = self.transfer_function.bode
+        self.bode_plot = self.transfer_function.bode_plot
+        self.mag_plot = self.transfer_function.mag_plot
```

Comment:
I don't see a good reason to add these methods dynamically and it certainly has disadvantages. (For instance, they won't show up in the API documentation.) So I'd rather implement them as proper proxy methods. On the other hand, I would prefer to completely get rid of them (after a deprecation cycle). 

I would also suggest to make `transfer_function` callable, so that one write `model.transfer_function(foo)`.

Reply:
> I don't see a good reason to add these methods dynamically and it certainly has disadvantages. (For instance, they won't show up in the API documentation.) So I'd rather implement them as proper proxy methods.

Ok, that makes sense.

> On the other hand, I would prefer to completely get rid of them (after a deprecation cycle).

I'm not sure about that. I guess it makes sense from the API design perspective, but, from the user perspective, I think that it would be nice to write `fom.bode_plot(...)` compared to `fom.transfer_function.bode_plot(...)`.

> I would also suggest to make `transfer_function` callable, so that one write `model.transfer_function(foo)`.

I thought about it, but then I was not sure what to do about `eval_dtf`. Only replacing `eval_tf` with `__call__` doesn't feel right to me.

Reply:
> > On the other hand, I would prefer to completely get rid of them (after a deprecation cycle).
> 
> I'm not sure about that. I guess it makes sense from the API design perspective, but, from the user perspective, I think that it would be nice to write `fom.bode_plot(...)` compared to `fom.transfer_function.bode_plot(...)`.

That may be true for the experienced user. But also from a user perspective I would argue that it can be confusing if there are different methods you can call which do the same thing. Another point is that each time you change something about the interface of `TransferFunction`, you have to reflect that in the `Model` (in particular adding/removing methods). So overall I would still be in favor of deprecating these methods. However, I see that this also somewhat a matter of taste, so I'd like to hear other opinions, @pymor/all ..

> 
> > I would also suggest to make `transfer_function` callable, so that one write `model.transfer_function(foo)`.
> 
> I thought about it, but then I was not sure what to do about `eval_dtf`. Only replacing `eval_tf` with `__call__` doesn't feel right to me.

Ok, but then maybe better `eval` and `eval_derivative`?

Reply:
> > > On the other hand, I would prefer to completely get rid of them (after a deprecation cycle).
> > 
> > 
> > I'm not sure about that. I guess it makes sense from the API design perspective, but, from the user perspective, I think that it would be nice to write `fom.bode_plot(...)` compared to `fom.transfer_function.bode_plot(...)`.
> 
> That may be true for the experienced user. But also from a user perspective I would argue that it can be confusing if there are different methods you can call which do the same thing. Another point is that each time you change something about the interface of `TransferFunction`, you have to reflect that in the `Model` (in particular adding/removing methods). So overall I would still be in favor of deprecating these methods.

Agreed. Its never good to have the same thing in multiple places.

Reply:
@lbalicki, what do you think?

Reply:
I think making it possible to use `fom.bode_plot(...)` vs `fom.transfer_function.bode_plot(...)` will only make things easier for users. If a user knows a little bit about systems theory it's not surprising at all that `fom` and `fom.transfer_function` have the exact same frequency response, bode plot, etc.. I would be for adding proxy methods for `bode_plot`, `freq_resp`, ... Or, if most other people are for deprecating these methods, we should at least specifically mention that `bode_plot`,... can be computed through `fom.transfer_function` in the according class documentations.

Reply:
If you, @pmli and @lbalicki are strongly convinced that these convenience methods are necessary, I will accept that, but I still think that it would be better to not have them. I have put the topic on today's dev meeting agenda. Let's decide then. (I would of course be totally fine with adding documentation in the Model ..)

Reply:
@pmli, have you come to a decision or do you want to postpone the pr after the release?

Reply:
@sdrave @lbalicki I pushed now the changes that deprecate the proxy methods. Let me know what you think (I didn't add the documentation yet, I'm not sure where it would go, maybe the attributes section?).

Reply:
Looks good! (BTW, if the deprecation turns out to be a mistake, we can, of course, add the methods again at a later time.) Regarding the documentation: I would just add a sentence to the main section of the class docstring. I believe @lbalicki had a nice formulation at the dev meeting. Something like 'methods related to the frequency domain, e.g. bode plots, ..., are attached to the `transfer_function` of the `Model`.'

Reply:
Ok. TBH, I still feel that having a class called something like `ModelWithTransferFunction` between `Model` and `LTIModel`/`SecondOrderModel`/`LinearDelayModel` would be nice. E.g., `hasattr(m, 'transfer_function')` could then be replaced with `isinstance(m, ModelWithTransferFunction)` and there would not be a need for `if instance(m, TransferFunction): m = m.transfer_function` in some places..

Reply:
Well I think a big advantage of the restructuring is that in many places where you actually only need the transfer function you can pass a `TransferFunction` instead of a `Model` that might not be available. Grepping through the branch, most of these checks seem to happen in places where I would argue that it is also okay to explicitly pass the transfer function instead of the `Model`. In particular, I'd actually force the user to write `fom1.transfer_function + transfer_function2` instead of `fom1 + transfer_function2`. But let's have that discussion after the release ..